### ISSUE TITLE
[Ideas / Discussion] Giant Feature Suggestion List

### ISSUE THREAD

**Issue by UnknownEvil** [evolvio/evolv.io#178]
I suggest there be more inputs, and ways to sense. 

I've noticed with some tweaking (added occlusion detection, but it's scrapped together for testing, so I won't be pull requesting) some creatures actually evolve a thing to only attack when another creature is near, but that makes it a bit difficult to reproduce.

So, I suggest they be able to tell if another creature is willing, and adapt if they can. And also, be able to detect where, or what direction a creature is looking. 

This certainly goes along with the past suggestions of scent, voice, etc. But, I do hope that we can come up with some nice, useful things. 

Some Feature Suggestions
1. choose the number of eyes that they have
1. more eyes = more energy
1. "looking" further costs more energy
* "looking" closer takes less
* or even having a base distance, and straying from that costs more energy, 
as someone (forgive me, but I can't remember) suggested
* emit sounds,
perhaps coming upon some sort of call system, although that's quite far-fetched.
* leave scent trails, and markings
* detect where a sound came from
* perhaps choose number of ears and how big (each ear has a left and right value)
* dynamically change vision distance, and angles.
* drinking water?
* evolve into species like water, land, and amphibious animals. 
Even different sizes.
(dunno how that'd work)
* FOV?
* Actually Fight
* Health
* Illness
* Aging
* Excrement Buildup (And a need to remove it, or loss of health)
* Flying creatures, if only you could tell.
* Digging
* Corpses

**Comment by PRabahy** [comment]
Aging is already in there.

I like idea 5 a lot. I was thinking about doing that for the distance. When a creature is born/created, it has a default distance that it can see. Its brain can then modify that distance, but the more it modifies it, the more energy it takes.

**Comment by UnknownEvil** [comment]
I was originally thinking, more distance, more energy, less distance, less energy. But at that point, all of the creatures would evolve to have all of their vision directly in the center of themselves, and that's it. So that is a much better option.

**Comment by UnknownEvil** [comment]
As a note: By aging I meant growtch over time, more susceptible to illness? Degrading of the senses, that kind of thing.

**Comment by NateUniAccount** [comment]
A load of great ideas there! With the creature changes, it all depends on how detailed we want to make the simulation. Ideally we'd want to add ways of evolving that can interact with each other so there'd be emergent behaviour.

**Comment by UnknownEvil** [comment]
Exactly, I want this to go somewhere, but it does seem like a lot of work, and perhaps it could all be to no avail.
